src/Microsoft.Performance.SDK.Runtime.NetCoreApp/Discovery/SandboxPreloadValidator.cs

```diff
@@ -0,0 +1,175 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Runtime.Serialization;
+using System.Security.Permissions;
+using Microsoft.Performance.SDK.Runtime.Discovery;
+using NuGet.Versioning;
+
+namespace Microsoft.Performance.SDK.Runtime.NetCoreApp.Discovery
+{
+    public sealed class SandboxPreloadValidator
+        : IPreloadValidator
+    {
+        private readonly MetadataAssemblyResolver resolver;
+        private readonly VersionChecker versionChecker;
+
+        public SandboxPreloadValidator(
+            IEnumerable<string> assemblyPaths,
+            VersionChecker versionChecker)
+        {
+            //
+            // https://docs.microsoft.com/en-us/dotnet/standard/assembly/inspect-contents-using-metadataloadcontext
+            // 
+            
+            var runtime = Directory.GetFiles(RuntimeEnvironment.GetRuntimeDirectory(), "*.dll");
+            var allDlls = assemblyPaths.Concat(runtime);
+
+            this.resolver = new PathAssemblyResolver(allDlls);
+            this.versionChecker = versionChecker;
+        }
+
+        public bool IsAssemblyAcceptable(string fullPath, out ErrorInfo error)
+        {
+            using (var context = new MetadataLoadContext(this.resolver))
```

Comment:
>using (var context = new MetadataLoadContext(this.resolver)) [](start = 12, length = 60)

This happens in a for loop. Might be better to just make SandboxPreloadValidator support IDisposable and keep the context around.